### PR TITLE
[zh-CN] Init FileSystemFileHandle

### PR DESCRIPTION
**Pull request by BSPR0002**
### Motivation

Init FileSystemFileHandle.
Update contents of File System API to sync with contents of the new translated.

### CODE REVIEW

File: files/zh-cn/web/api/file_system_api/index.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ slug: Web/API/File_System_API
 - {{domxref('HTML Drag and Drop API', 'HTML 拖放 API', '', 'nocode')}} 的 {{domxref('DataTransferItem.getAsFileSystemHandle()')}} 方法。
 - [文件处理 API](https://developer.chrome.com/en/articles/file-handling/)。
 
-每种句柄都提供了其独有的功能，取决于你使用的种类，会有些许差异（详见[接口](#接口)部分）。在获得句柄后，你便可以访问文件的数据或是被选中的目录的信息（包含子目录）。此 API 开辟了 web 此前一直缺乏的潜在功能。但不论如何，安全性是设计 API 时的首要考量，除非用户明确授权，否则就不允许访问文件和目录的数据。
+每种句柄都提供了其独有的功能，取决于你使用的种类，会有些许差异（详见[接口](#接口)部分）。在获得句柄后，你便可以访问文件的数据或是被选中的目录的信息（包含子目录）。此 API 开辟了 web 此前一直缺乏的潜在功能。但不论如何，安全性是设计 API 时的首要考量，除非用户明确授权，否则就不允许访问文件和目录的数据（注意：源私有文件系统并非如此，因为其对用户不可见）。
```
Comment (yin1999) [inline files/zh-cn/web/api/file_system_api/index.md:23]:
```suggestion
每种句柄都提供了其独有的功能，取决于你使用的种类，会有些许差异（详见[接口](#接口)部分）。在获得句柄后，你便可以访问文件的数据或是被选中的目录的信息（包含子目录）。此 API 开辟了 web 此前一直缺乏的潜在功能。但不论如何，安全性是设计 API 时的首要考量，除非用户明确授权，否则就不允许访问文件和目录的数据（注意：[源私有文件系统](#源私有文件系统)并非如此，因为其对用户不可见）。
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [文件系统 API](/zh-CN/docs/Web/API/File_System_Access_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:67]:
```suggestion
- [文件系统 API](/zh-CN/docs/Web/API/File_System_API)
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [文件系统 API](/zh-CN/docs/Web/API/File_System_Access_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:64]:
```suggestion
- [文件系统 API](/zh-CN/docs/Web/API/File_System_API)
```

File: files/zh-cn/web/api/filesystemfilehandle/getfile/index.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [文件系统 API](/zh-CN/docs/Web/API/File_System_Access_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/getfile/index.md:55]:
```suggestion
- [文件系统 API](/zh-CN/docs/Web/API/File_System_API)
```

File: files/zh-cn/web/api/filesystemfilehandle/index.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [文件系统 API](/zh-CN/docs/Web/API/File_System_Access_API)
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/index.md:127]:
```suggestion
- [文件系统 API](/zh-CN/docs/Web/API/File_System_API)
```

File: files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+---
+title: FileSystemFileHandle：createSyncAccessHandle() 方法
+slug: Web/API/FileSystemFileHandle/createSyncAccessHandle
+---
+
+{{securecontext_header}}{{APIRef("File System Access API")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createsyncaccesshandle/index.md:6]:
```suggestion
{{securecontext_header}}{{APIRef("File System API")}}
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
+---
+title: FileSystemFileHandle：createWritable() 方法
+slug: Web/API/FileSystemFileHandle/createWritable
+---
+
+{{securecontext_header}}{{APIRef("File System Access API")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:6]:
```suggestion
{{securecontext_header}}{{APIRef("File System API")}}
```

File: files/zh-cn/web/api/filesystemfilehandle/getfile/index.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
+---
+title: FileSystemFileHandle：getFile() 方法
+slug: Web/API/FileSystemFileHandle/getFile
+---
+
+{{securecontext_header}}{{APIRef("File System Access API")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/getfile/index.md:6]:
```suggestion
{{securecontext_header}}{{APIRef("File System API")}}
```

File: files/zh-cn/web/api/filesystemfilehandle/index.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
+---
+title: FileSystemFileHandle
+slug: Web/API/FileSystemFileHandle
+---
+
+{{securecontext_header}}{{APIRef("File System Access API")}}
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/index.md:6]:
```suggestion
{{securecontext_header}}{{APIRef("File System API")}}
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+{{domxref("FileSystemFileHandle")}} 接口的 **`createWritable()`** 方法用于创建一个 {{domxref('FileSystemWritableFileStream')}} 对象，可用于写入文件。此方法返回一个可兑现这些写入流的 {{jsxref('Promise')}} 对象。
+
+任何通过写入流造成的更改在写入流被关闭前都不会反映到文件句柄所代表的文件上。这通常是将数据写入到一个临时文件来实现的，然后只有在写入流被关闭后才会用临时文件替换掉文件句柄所代表的文件。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:10]:
```suggestion
任何通过写入流造成的更改在写入流被关闭前都不会反映到文件句柄所代表的文件上。这通常是将数据写入到一个临时文件来实现的，然后只有在写入文件流被关闭后才会用临时文件替换掉文件句柄所代表的文件。
```

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+### 参数
+
+- `options` {{optional_inline}}
+
+  - : 一个包含以下属性的对象：
+
+    - `keepExistingData`
+      - : {{jsxref('Boolean', '布尔值')}}，默认为 `false`。当设为 `true` 时，如果文件存在，则现将现有文件的内容复制到临时文件，否则临时文件初始时内容为空。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md:26]:
```suggestion
      - : {{jsxref('Boolean', '布尔值', '', 'nocode')}}，默认为 `false`。当设为 `true` 时，如果文件存在，则现将现有文件的内容复制到临时文件，否则临时文件初始时内容为空。
```

File: files/zh-cn/web/api/filesystemfilehandle/index.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+
+{{domxref("File System Access API", "File System Access API", "", "nocode")}} 的 **`FileSystemFileHandle`** 接口表示了一个指向文件系统条目的句柄。此接口通过 {{domxref('window.showOpenFilePicker()')}} 方法来访问。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/index.md:8]:
```suggestion
{{domxref("File System API", "File System API", "", "nocode")}} 的 **`FileSystemFileHandle`** 接口表示一个指向文件系统条目的句柄。可通过 {{domxref('window.showOpenFilePicker()')}} 方法来访问此接口。
```

File: files/zh-cn/web/api/filesystemfilehandle/index.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+注意，读写操作所依赖的文件访问权限在刷新或关闭了页面并且页面所属的源没有其他标签页保持打开的情况下不会继续保有。{{domxref("FileSystemHandle")}} 接口的 {{domxref("FileSystemHandle.queryPermission()", "queryPermission")}} 方法可用来在访问文件前验证权限状态。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/index.md:10]:
```suggestion
注意，读写操作所依赖的文件访问权限在刷新或关闭页面并且页面所属的源没有其他标签页保持打开的情况下不会继续保有。{{domxref("FileSystemHandle")}} 接口的 {{domxref("FileSystemHandle.queryPermission()", "queryPermission")}} 方法可用于在访问文件前验证权限状态。
```

File: files/zh-cn/web/api/filesystemfilehandle/index.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+
+  // 将更改持久化至磁盘
+  accessHandle.flush();
+
+  // 用完 FileSystemSyncAccessHandle 记得把它关闭
+  accessHandle.close();
+};
+```
+
+> **备注：** 在规范早期版本中，{{domxref("FileSystemSyncAccessHandle")}} 的 {{domxref("FileSystemSyncAccessHandle.close()", "close()")}}、{{domxref("FileSystemSyncAccessHandle.flush()", "flush()")}}、{{domxref("FileSystemSyncAccessHandle.getSize()", "getSize()")}} 和 {{domxref("FileSystemSyncAccessHandle.truncate()", "truncate()")}} 被不合逻辑地指定为异步方法。此问题现已被[修正](https://github.com/whatwg/fs/issues/7)，但某些浏览器依然支持异步版本。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemfilehandle/index.md:115]:
```suggestion
> **备注：** 在规范早期版本中，{{domxref("FileSystemSyncAccessHandle.close()", "close()")}}、{{domxref("FileSystemSyncAccessHandle.flush()", "flush()")}}、{{domxref("FileSystemSyncAccessHandle.getSize()", "getSize()")}} 和 {{domxref("FileSystemSyncAccessHandle.truncate()", "truncate()")}} 被错误地定义为异步方法。此问题现已被[修正](https://github.com/whatwg/fs/issues/7)，但某些浏览器依然支持异步版本。
```

**Review by yin1999 — COMMENTED** [review]
some suggestions. Maybe you should update your local `mdn/content` repo

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md (hunk 11 above)
Comment (BSPR0002) [reply files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md]: Is chinese title links should be formated to 'nocode'?

**Review by BSPR0002 — COMMENTED** [review]
This branch was created on July 16th, and after completing some of the content, I put it on hold. After that, the File System API was updated, so there were some omissions in the subsequent modifications.

File: files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md (hunk 11 above)
Comment (yin1999) [reply files/zh-cn/web/api/filesystemfilehandle/createwritable/index.md]: Yes. It looks better than warpped it with `code`

**Review by yin1999 — APPROVED** [review]
Thank you @BSPR0002 :tada:

**Pull request merged**